crates/typst/src/foundations/calc.rs

```diff
@@ -977,3 +1186,11 @@ cast! {
 fn too_large() -> &'static str {
     "the result is too large"
 }
+
+/// The hinted error message when trying to apply an operation to decimal and
+/// float operands.
+#[cold]
+fn cant_apply_to_decimal_and_float() -> HintedString {
+    HintedString::new(eco_format!("cannot apply this operation to a decimal and a float"))
```

Comment:
In principle I didn't feel like including the operation name explicitly here, but let me know if that would be desired